### ISSUE TITLE
1.6.0 contains breaking changes

### ISSUE THREAD

**Issue by fascinosum** [colinmollenhour/php-redis-session-abstract#58]
Hello,

A new minor release of the project contains a [breaking change](https://github.com/colinmollenhour/php-redis-session-abstract/compare/v1.5.5...v1.6.0) to the interface that, according to Semantic Versioning, should result in a new major release. This breaking change significantly affects other open-source projects, such as [Magento OS](https://github.com/magento/magento2), which has thousands of users
@colinmollenhour, could you kindly consider reverting the change to the interface and releasing a new patch version for 1.6.*?

Thank you, @colinmollenhour, and all contributors of the project for your hard work!

**Comment by hostep** [comment]
Indeed, I agree

In M2, we now get this when running `bin/magento setup:di:compile`:

```
BPHP Fatal error:  Class Magento\Framework\Session\SaveHandler\Redis\Config contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (Cm\RedisSession\Handler\ConfigInterface::getRetries) in vendor/magento/framework/Session/SaveHandler/Redis/Config.php on line 16

Fatal error: Class Magento\Framework\Session\SaveHandler\Redis\Config contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (Cm\RedisSession\Handler\ConfigInterface::getRetries) in vendor/magento/framework/Session/SaveHandler/Redis/Config.php on line 16
```

Because https://github.com/magento/magento2/blob/2.4-develop/lib/internal/Magento/Framework/Session/SaveHandler/Redis/Config.php implements the interface which was now changed in 1.6.0, so this breaks.

**Comment by cmuench** [comment]
It's breaking Magento 2.4.6-p5 as welll.
For reference: https://github.com/netz98/n98-magerun2/actions/runs/9119272478/job/25087515280

Maybe this change should be tagged as v2.0.0.

**Comment by colinmollenhour** [comment]
Yikes, sorry for the trouble, guys! I looked at the M2 composer.json and it had the version as "^1.5" and I thought that meant ">=1.5.0 <1.6.0" but apparently it means ">=1.5.0 <2.0.0".. I will move this change to 2.0

**Comment by colinmollenhour** [comment]
I released the latest master as v2.0.0 and the previous v1.5.5 as v1.7 so I believe that should effectively "roll back" the v1.6.0 release.

**Comment by hostep** [comment]
Thank you very much for the quick fix Colin! And don't worry too much about the mistake, we're only human after all 🙂
